src/Components/ObjectProperties/ObjectProperties.js

```diff
@@ -46,12 +46,18 @@ class ObjectProperties extends React.Component {
     super(props);
 
     this.state = {
-      keyWidth: null,
+      keyWidth: this.props.keyWidth || null,
     };
 
     this.keyWidthsIndex = {};
   };
 
+  componentWillReceiveProps = (nextProps) => {
+    if (nextProps.keyWidth) {
+      this.setState({keyWidth: nextProps.keyWidth});
```

Comment:
I am afraid this is an anti pattern. You might potentially trigger 2 re-renders with this code. If you want to set an initial state, do that from the `constructor` function.